tests/components/tradfri/test_cover.py

```diff
@@ -0,0 +1,219 @@
+"""Tradfri cover (recognised as blinds in the IKEA echosystem) platform tests."""
+
+from copy import deepcopy
+from unittest.mock import MagicMock, Mock, PropertyMock, patch
+
+import pytest
+from pytradfri.device import Device
+from pytradfri.device.blind import Blind
+from pytradfri.device.blind_control import BlindControl
+
+from homeassistant.components import tradfri
+
+from . import GATEWAY_ID
+
+from tests.common import MockConfigEntry
+
+SET_POSITION_TEST_CASES = [
+    # Open cover
+    [{}, {"position": 100}, {"state": "open"}],
+    # Close cover
+    [{}, {"position": 0}, {"state": "closed"}],
+]
+
+
+@pytest.fixture(autouse=True, scope="module")
+def setup(request):
+    """Set up patches for pytradfri methods."""
+    p_1 = patch(
+        "pytradfri.device.BlindControl.raw",
+        new_callable=PropertyMock,
+        return_value=[{"mock": "mock"}],
+    )
+    p_2 = patch("pytradfri.device.BlindControl.blinds")
+    p_1.start()
+    p_2.start()
+
+    def teardown():
+        """Remove patches for pytradfri methods."""
+        p_1.stop()
+        p_2.stop()
+
+    request.addfinalizer(teardown)
```

Comment:
We can yield in the fixture instead and then just call stop on the patches. Or use a context manager, ie a with statement, for the patches and yield inside the context manager.

Example:
https://github.com/home-assistant/core/blob/a978661545192928bcc994984e963cdb7b3b2bf7/tests/components/wled/conftest.py#L27-L31

Reply:
Have have changed this into what I believe you meant me to do here. Do you want me to do this in the light test-file as well?

Reply:
Yeah, but let's do that in a separate PR. There may be more things to clean up there.